diff --git a/bootstrap/bootstrap b/bootstrap/bootstrap
index fc1d77606d0..3631965ce4a 100755
--- a/bootstrap/bootstrap
+++ b/bootstrap/bootstrap
@@ -1112,9 +1112,9 @@ sunpro)
 	echo "CPP=			\${CC} -E" >> ${TARGET_MKCONF}
 	;;
 clang)
-	echo "CC=			clang"     >> ${TARGET_MKCONF}
-	echo "CXX=			clang++"   >> ${TARGET_MKCONF}
-	echo "CPP=			\${CC} -E" >> ${TARGET_MKCONF}
+	echo "CC=			${CC:-clang}"        >> ${TARGET_MKCONF}
+	echo "CXX=			${CXX:-clang++}"     >> ${TARGET_MKCONF}
+	echo "CPP=			${CPP:-\$\{CC\} -E}" >> ${TARGET_MKCONF}
 	if [ -z "$CLANGBASE" -a -f "/usr/bin/clang" ]; then
 		CLANGBASE="/usr"
 	fi
